more_itertools/more.py

```diff
@@ -1106,3 +1108,83 @@ def always_iterable(obj):
         return obj,
 
     return obj
+
+def adjacent(predicate, iterable, distance=1):
+    """
+    Returns an iterable over ``(bool, item)`` tuples where the ``item`` is drawn
+    from the underlying *iterable* and the ``bool`` indicates whether that item
+    satisfies the *predicate* or is adjacent to one that does. For example:
+
+        >>> list(adjacent(lambda x: x % 4 == 2, range(6)))
+        [(False, 0), (True, 1), (True, 2), (True, 3), (False, 4), (False, 5)]
+
+    The *distance* used to determine what counts as adjacent can be configured:
+
+        >>> list(adjacent(lambda x: x % 4 == 2, range(6), 2))
+        [(True, 0), (True, 1), (True, 2), (True, 3), (True, 4), (False, 5)]
+
+    In this case any elements within 2 positions of the one that satisfies the
+    predicate are paired with ``True``. *distance* can be any nonnegative
```

Comment:
Could you break a paragraph after the "In this case..." sentence? Also, I'd probably leave out the example of this being the same as `map` and `zip`; if the reader wants to prove it to themselves we shouldn't deprive them of the exercise.